src/sync-external-store--stateful-store/store.ts

```diff
@@ -0,0 +1,44 @@
+import { SyncExternalStore } from "../sync-external-store/index.js";
+import {
+  CreateErrorSnapshot,
+  CreateLoadingSnapshot,
+  CreateSuccessSnapshot,
+  createErrorSnapshot,
+  createLoadingSnapshot,
+  createSuccessSnapshot,
+} from "./create-snapshot.js";
+import { StoreState } from "./store-item.js";
+
+type UpdateStoreParams<TStore extends StoreState, TParams = void, TResponse = void> = {
+  key: keyof TStore;
+  params?: TParams;
+  service: (params: TParams) => Promise<TResponse>;
+  updateSnapshot?: {
+    loading: CreateLoadingSnapshot;
+    success: CreateSuccessSnapshot;
+    error: CreateErrorSnapshot;
+  };
+};
+
+export abstract class StatefulSyncExternalStore<TStore extends StoreState> extends SyncExternalStore<TStore> {
```

Comment:
Can you create unit tests to help document usage? Also you know, code coverage?

Reply:
I pulled this down... in the original context-store there were three store types(copied from the readme):

- useContextStore - For non-indexable stores like objects and primatives or when you don't need to edit individual elements
- useIndexableContextStore - For indexable stores like maps or arrays with a single loading state at the root
- useIndexableStatefulContextStore - For indexable stores but when you want to maintain separate load states per item than the list of items

From the StoreState object you created, this looks to be specifically IndexableStateful

It seems it might make sense to have three classes then:
 - StatefulSyncExternalStore
 - IndexableStatefulSyncExternalStore
 - IndexableItemStatefulSyncExternalStore
 
 Or something...